File: Data Structures & Algorithms/jump-game-ii/submission-2.java
class Solution {
    int dfs(int i, int[] nums){
        if(i>=nums.length) return 100000;
        if(i==nums.length-1) return 0;
        int res = 100000000 ;
        for(int  j=1 ; j<= nums[i] ;j++){
            res=Math.min(res,dfs(i+j,nums)+1);
        }
        return res ;
    }
    public int jump(int[] nums) {
        return dfs(0,nums);
    }
}

File: Data Structures & Algorithms/greatest-common-divisor-traversal/submission-2.py
class Solution:
    def canTraverseAllPairs(self, nums: List[int]) -> bool:
        n=len(nums)
        mp={}
        for i in range(n-1):
            if nums[i]==1 :
                return False
            if nums[i] not in mp:
                mp[nums[i]]=set()
            for j in range(i+1,n):
                if math.gcd(nums[i],nums[j])>1 :
                    mp[nums[i]].add(nums[j])
                    if nums[j] not in mp:
                        mp[nums[j]]={nums[i]}
                    else:
                        mp[nums[j]].add(nums[i])
        visited=set()
        def dfs(num) :
            visited.add(num)
            for nu in mp[num] :
                if nu not in visited :
                    dfs(nu)
        dfs(nums[0])
        return len(visited)==len(set(nums)) 